client/src/lib/dependency-injection/buildApolloGraphqlContainer.js

```diff
@@ -0,0 +1,162 @@
+import buildBaseContainer from './buildBaseContainer';
+import { captureTag } from './graphql/tags';
+import ApolloGraphqlManager from './ApolloGraphqlManager';
+import buildCreateMutation from './graphql/buildCreateMutation';
+import buildReadQuery from './graphql/buildReadQuery';
+import buildReadOneQuery from './graphql/buildReadOneQuery';
+import buildUpdateMutation from './graphql/buildUpdateMutation';
+import buildDeleteMutation from './graphql/buildDeleteMutation';
+import buildBaseQuery from './graphql/buildBaseQuery';
+import { CREATE, READ, READ_ONE, UPDATE, DELETE, QUERY } from './graphql/templates';
+
+const buildApolloGraphqlContainer = (base = buildBaseContainer()) => ({
+  ...base,
+
+  /**
+   * A register of templates registered with this container
+   */
+  templates: {
+    [CREATE]: buildCreateMutation(captureTag),
+    [READ]: buildReadQuery(captureTag),
+    [READ_ONE]: buildReadOneQuery(captureTag),
+    [UPDATE]: buildUpdateMutation(captureTag),
+    [DELETE]: buildDeleteMutation(captureTag),
+    [QUERY]: buildBaseQuery(captureTag),
```

Comment:
We should initialise with these empty and inflate them at boot time. Would help to decouple Injector from SilverStripe. Each consumer of Injector should in theory be supplying its own set of templates.

Reply:
Can you implement this or should we leave it for this merge @unclecheese ?

Reply:
done